src/internal/_Set.js

```diff
@@ -37,11 +37,15 @@ module.exports = (function() {
       case 'string':
       case 'number':
         // distinguish between +0 and -0
-        if (item === 0 && !set._items['-0'] && 1 / item === -Infinity) {
-          if (shouldAdd) {
-            set._items['-0'] = true;
+        if (item === 0 && 1 / item === -Infinity) {
+          if (set._items['-0'] === true) {
+            return !shouldAdd;
+          } else {
+            if (shouldAdd) {
+              set._items['-0'] = true;
+            }
+            return shouldAdd;
           }
```

Comment:
> I find the logic above difficult to follow.

I agree. I think the confusion is particularly caused by the overloaded boolean return value. I've updated the PR such that the function will now just return `true` or `false` based on whether the value previously existed.

> I would also like to see {'-0': false} rather than {} as the initial value of set._items. This would remove the need for === true, and would indicate—in the constructor—the purpose of the _items property.

The original PR had this declared explicitly, though we decided to move them all to the same mechanism for consistency. I can add them all back if desired.